===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(p11_asn1_defs_load:defs)) at 12 (line 87, column 8),
        Insert: if ({ p11_asn1_defs_load:n$8 == { 0 } }) free(*(p11_asn1_defs_load:defs->buckets)) at 11 (line 87, column 8) }
===== Actual patch in diff format =====
@@ -85,6 +85,8 @@ p11_asn1_defs_load (void)
 		}
 
 		if (!p11_dict_set (defs, (void *)asn1_tabs[i].prefix, def))
+			free(defs->buckets);
+			free(defs);
 			return_val_if_reached (NULL);
 	}
 
